metric/collector/policy_count.go

```diff
@@ -0,0 +1,86 @@
+package collector
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/bytebase/bytebase/api"
+	metricAPI "github.com/bytebase/bytebase/metric"
+	"github.com/bytebase/bytebase/plugin/metric"
+	"github.com/bytebase/bytebase/plugin/metric/collector"
+	"github.com/bytebase/bytebase/store"
+	"go.uber.org/zap"
+)
+
+// policyCollector is the metric data collector for policy.
+type policyCollector struct {
+	l     *zap.Logger
+	store *store.Store
+}
+
+// NewPolicyCollector creates a new instance of policyCollector
+func NewPolicyCollector(l *zap.Logger, store *store.Store) collector.MetricCollector {
+	return &policyCollector{
+		l:     l,
+		store: store,
+	}
+}
+
+// Collect will collect the metric for policy
+func (c *policyCollector) Collect(ctx context.Context) ([]*metric.Metric, error) {
+	var res []*metric.Metric
+
+	policyList, err := c.store.ListPolicy(ctx, &api.PolicyFind{})
+	if err != nil {
+		return nil, err
+	}
+
+	policyCountMap := make(map[string]metricAPI.PolicyCountMetric)
+
+	for _, policy := range policyList {
+		key := fmt.Sprintf("%s_%s", policy.Type, policy.Environment.Name)
```

Comment:
If using key, then environment.id can guarantee uniqueness

Reply:
Of course, but seems the env name is set to be the unique key in our DB

Reply:
Oh, forgot that